File: Python/Strings/2062 - OBI URI.py
"""
Autor: Pierre Vieira
Data da submissão: 26/01/2020 10:41:23
"""


def conserta_palavra(palavra):
    if len(palavra) != 3:
        return palavra
    elif palavra[0:2] == 'OB':
        return 'OBI'
    elif palavra[0:2] == 'UR':
        return 'URI'
    return palavra


n = int(input())
palavras = input().split()
palavras = tuple(map(conserta_palavra, palavras))
for c in range(len(palavras)-1):
    print(palavras[c], end=' ')
print(palavras[len(palavras) - 1])

File: C/Strings/2062 - OBI URI.c
/*
 * Autor: Pierre Vieira
 * Data de submissão: 26/01/2020 10:55:02
 */

#include <stdio.h>
#include <string.h>

void faz_verificacao(char *palavra){
    int tamanho_palavra;
    tamanho_palavra = strlen(palavra);
    if (tamanho_palavra != 3){
        printf("%s", palavra);
    }
    else if(palavra[0] == 'O' && palavra[1] == 'B' && palavra[2] != 'I'){
        printf("OBI");
    }
    else if(palavra[0] == 'U' && palavra[1] == 'R' && palavra[2] != 'I'){
        printf("URI");
    }
    else{
        printf("%s", palavra);
    }
}

int main(){
    int n;
    char palavra[20];
    scanf("%d", &n);
    while (n > 1){
        scanf("%s", palavra);
        faz_verificacao(palavra);
        printf(" ");
        n--;
    }
    scanf("%s", palavra);
    faz_verificacao(palavra);
    printf("\n");
    return 0;
}